lm_optimizer.py

```diff
@@ -28,11 +28,23 @@ def objective(trial):
     FLAGS.lm_alpha = trial.suggest_uniform('lm_alpha', 0, FLAGS.lm_alpha_max)
     FLAGS.lm_beta = trial.suggest_uniform('lm_beta', 0, FLAGS.lm_beta_max)
 
-    tfv1.reset_default_graph()
-    samples = evaluate(FLAGS.test_files.split(','), create_model)
-
     is_character_based = trial.study.user_attrs['is_character_based']
 
+    samples = []
+    for step, test_file in enumerate(FLAGS.test_files.split(',')):
+        tfv1.reset_default_graph()
+
+        current_samples = evaluate([test_file], create_model, try_loading)
+        samples += current_samples
+
+        # Report intermediate objective value.
+        wer, cer = wer_cer_batch(current_samples)
+        trial.report(cer if is_character_based else wer, step)
+
+        # Handle pruning based on the intermediate value.
+        if trial.should_prune():
```

Comment:
Does the study actually have a Pruner? If so is it not NopPruner?

Reply:
Seems like it's [MedianPruner](https://github.com/optuna/optuna/blob/master/optuna/study.py#L200). I think it should work.

The only thing that gives me pause is the fact that different csv's in FLAGS.test_files may require very different alpha+beta. Thus the variance of wer and cer will be large. But if the [MedianPruner](https://github.com/optuna/optuna/blob/fd487d4c3ec7cba11ee84c103efd5845937f9789/optuna/pruners/median.py#L4) acts as advertised, i.e. "prunes if the trial's best intermediate result is worse than median of intermediate results of previous trials **at the same step**", then it should be OK.

The important point here being "...at the same step" and Python guaranteeing a consistent ordering from the split() call above.